.github/workflows/actions/ubuntu-build-deps/action.yml

```diff
@@ -0,0 +1,50 @@
+name: 'ubuntu build dependencies'
+description: 'install required build dependencies for ubuntu'
+inputs:
+
+  SUDO:
+    description: "set to true to run apt as root"
+    required: false
+    default: ""
+
+
+runs:
+  using: 'composite'
+  steps:
+    - name: apt update
+      shell: bash
+      run: ${{inputs.SUDO}} apt update
+
+    - name: install build tools
+      shell: bash
+      run: |
+        ${{inputs.SUDO}} apt install -y \
+            build-essential \
+            clang \
+            clang-tidy \
+            ninja-build \
+            iwyu \
+            pkgconf \
+            wget 
+        if [ ${{ matrix.os }} != 'ubuntu:16.04' ]; then apt install -y clang-tools; fi
+
+    - name: install coturn dependencies
+      shell: bash
+      run: |
+        ${{inputs.SUDO}} apt install -y \
+            libevent-dev \
+            libssl-dev \
+            libpq-dev \
+            libsqlite3-dev \
+            libhiredis-dev \
+            libmongoc-dev \
+            libmicrohttpd-dev
+        if [ ${{ matrix.os }} = 'ubuntu:16.04' ]; then apt-get install -y libmariadb-client-lgpl-dev; fi
```

Comment:
Apparently github allows composite actions to pick up matrix details from the targets that pull it in?

that's cool

But be careful here, it's not guaranteed that the composite action will be used by another job that uses a matrix.

Probably better is to use `if [ "$(lsb_release -s -r)x" == "16.04x" ]`

to determine if the runtime environment itself thinks that it's running inside ubuntu 16.04.

Reply:
Also, definitely assume my bash-foo is wrong, with my code-snippet above. Make sure to check google / stackoverflow on how to properly do string comparison in bash.